pkg/ccl/changefeedccl/kvfeed/kv_feed.go

```diff
@@ -214,7 +214,14 @@ func (f *kvFeed) run(ctx context.Context) (err error) {
 		boundaryType := jobspb.ResolvedSpan_BACKFILL
 		if f.schemaChangePolicy == changefeedbase.OptSchemaChangePolicyStop {
 			boundaryType = jobspb.ResolvedSpan_EXIT
-		} else if events, err := f.tableFeed.Peek(ctx, highWater.Next()); err == nil && isPrimaryKeyChange(events) {
+		} else if events, err := f.tableFeed.Peek(ctx, highWater.Next()); err == nil && isRegionalByRowChange(events) {
+			// NOTE(ssd): The user is unlikely to see this
+			// error. The schemafeed will fail with an
+			// non-retriable error, meaning we likely
+			// return right after runUntilTableEvent
+			// above.
```

Comment:
Note: because of this, if we want the absolutely most minimal change, we can avoid any changes to the event_table_filter and this code and just rely on the error coming out of validate.  I realized that the validate changes were  enough after writing this code and then thought it might be good to keep it in case the validate code changes, but given that we have a test for in-flight changefeeds failing, perhaps we can drop it.

Reply:
I've split this into a second commit to make it easier to drop if we want to.